### PR TITLE
Feat add AWS modules

### PR DESCRIPTION
**Pull request by HeverFarber**
* Those modules contain full-blown EKS installation.
* Readme - how to use it and also how to install the agent

### out of scope - the creation of secret and policy for the secret manager. it depends on Avenr PRD

### CODE REVIEW

**Comment by HeverFarber** [comment]
[K8S agent docs - Add complete TF stack for preparing for env0 agent installation - AWS](https://trello.com/c/r1UsBfoC/4857-k8s-agent-docs-add-complete-tf-stack-for-preparing-for-env0-agent-installation-aws)

File: aws/vpc/main.tf [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  private_subnet_tags = {
+    "kubernetes.io/cluster/${var.cluster_name}" = "shared"
+    "kubernetes.io/role/internal-elb"           = "1"
+    "tier"                                      = "private"
+  }
+}
+
+locals {
+  // zones supported by our kind of machine type in our region
+  aws_availability_zones = ["us-east-1a", "us-east-1b", "us-east-1c", "us-east-1d", /*"us-east-1e",*/ "us-east-1f"]
```
Comment (razbensimon) [inline aws/vpc/main.tf]: why `us-east-1e` is in comment? remove or add it

File: aws/variables.tf [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  description = "List of private subnets inside the VPC"
+
+  default = ["172.16.0.0/21", "172.16.16.0/21", "172.16.32.0/21", "172.16.48.0/21", "172.16.64.0/21"]
+}
+
+variable "public_subnets" {
+  description = "List of public subnets inside the VPC"
+
+  default = ["172.16.8.0/22", "172.16.24.0/22", "172.16.40.0/22", "172.16.56.0/22", "172.16.72.0/22"]
+}
```
Comment (razbensimon) [inline aws/variables.tf:38]: those are the same as written in `vpc` variables file.
We should remove the duplication or at least share it somehow.

File: README.md [hunk 3]
```diff
@@ -1 +1,81 @@
...
+      command     = "aws"
+    }
+  }
+}
+```
+#### cluster.tf
+Full-blown install
+```terraform
+module "my-cluster" {
+  source = "github.com/env0/k8s-modules//aws"
```
Comment (razbensimon) [inline README.md:54]: why duplicate slash? `//aws

File: README.md [hunk 4]
```diff
@@ -1 +1,81 @@
...
+
+  region       = var.region
+  cluster_name = var.cluster_name
+}
+``` 
+#### csi-driver.tf
+Specific module - see AWS folder for other modules
+```terraform
+module "csi-driver" {
+  source = "github.com/env0/k8s-modules//aws/csi-driver"
```
Comment (razbensimon) [inline README.md:66]: also here

**Review by razbensimon — COMMENTED** [review]
LGTM but I don't really have the confidence and specific knowledge for that approval. I think it would be nice if someone else will join for CR here. 🙏🏼

**Comment by razbensimon** [comment]
@env0/pr-reviewers

**Comment by RLRabinowitz** [comment]
Thanks @razbensimon I'll take a look

File: README.md [hunk 5]
```diff
@@ -1 +1,81 @@
 # k8s-modules
+This repository contains live examples for creation of kubernetes cluster which can run env0 deployments.
+you can take those examples and use them as-is or fork the repo and adjustment for yourself.
```
Comment (RLRabinowitz) [inline README.md]:
```suggestion
you can take those examples and use them as-is or fork the repo and adjust for yourself.
```

File: aws/vpc/main.tf (hunk 1 above)
Comment (HeverFarber) [reply aws/vpc/main.tf]: I took all files from our repo - I think this zone doesn't support this server type. I will remove it

File: aws/variables.tf (hunk 2 above)
Comment (HeverFarber) [reply aws/variables.tf:38]: TF doesn't know to take default from modules 
So - if someone will use this specific or from full blown he needs it

File: README.md (hunk 3 above)
Comment (HeverFarber) [reply README.md:54]: TF style - to say folder

File: README.md (hunk 4 above)
Comment (HeverFarber) [reply README.md:66]: same

File: README.md [hunk 6]
```diff
@@ -1 +1,81 @@
...
+
+you can build a mix of the modules (csi-driver) if you like
```
Comment (RLRabinowitz) [inline README.md]:
```suggestion
You can, of course, mix and match according to your own needs.
For example, you can use your own existing EKS cluster here, but then you'd need to add `data`s where necessary when replacing `module.eks` usages
```

File: README.md [hunk 7]
```diff
@@ -1 +1,81 @@
...
+
+### AWS 
+#### providers.tf
```
Comment (RLRabinowitz) [inline README.md]:
```suggestion
#### Setting up providers
Create a `providers.tf` file with the following content
```

File: README.md [hunk 8]
```diff
@@ -1 +1,81 @@
...
+    exec {
+      api_version = "client.authentication.k8s.io/v1alpha1"
+      args        = ["eks", "get-token", "--cluster-name", data.aws_eks_cluster.cluster.name]
+      command     = "aws"
+    }
+  }
+}
+```
+#### cluster.tf
+Full-blown install
```
Comment (RLRabinowitz) [inline README.md]:
```suggestion
#### Creating a full-blown cluster installation
If you'd like to create a cluster from scratch, including a VPC and an EFS for storage, you can simply use the `aws` root folder as a module
```

File: README.md [hunk 9]
```diff
@@ -1 +1,81 @@
...
+```terraform
+module "my-cluster" {
+  source = "github.com/env0/k8s-modules//aws"
+
+  region       = var.region
+  cluster_name = var.cluster_name
+}
+``` 
+#### csi-driver.tf
+Specific module - see AWS folder for other modules
```
Comment (RLRabinowitz) [inline README.md]:
```suggestion
#### Partial installation
You can also just pick out the parts necessary for your installation.
For example, only create a CSI-driver for storage:
```

File: README.md [hunk 10]
```diff
@@ -1 +1,81 @@
...
+  name             = "env0-agent"
+  namespace        = "env0-agent"
+  chart            = "env0-agent"
+  create_namespace = true
+  repository       = "https://env0.github.io/self-hosted"
+  timeout          = 600
+  values           = [
+    yamlencode(merge(jsondecode(var.env0_values), jsondecode(var.customer_values)))
+  ]
+}
```
Comment (RLRabinowitz) [inline README.md]:
```suggestion
#### Installing the env0 agent
If you'd like, you can also install the env0 agent using terraform.
Make sure you fill out `var.env0_values` according to the values provided to you by env0
```terraform
resource "helm_release" "agent" {
  depends_on       = [module.my-cluster] // add all necessary dependencies here 
  name             = "env0-agent"
  namespace        = "env0-agent"
  chart            = "env0-agent"
  create_namespace = true
  repository       = "https://env0.github.io/self-hosted"
  timeout          = 600
  values           = [
    yamlencode(merge(jsondecode(var.env0_values), jsondecode(var.customer_values)))
  ]
}
```

File: README.md [hunk 11]
```diff
@@ -1 +1,81 @@
...
+resource "helm_release" "agent" {
+  depends_on       = [module.my-cluster] // add all dependency 
+  name             = "env0-agent"
+  namespace        = "env0-agent"
+  chart            = "env0-agent"
+  create_namespace = true
+  repository       = "https://env0.github.io/self-hosted"
+  timeout          = 600
+  values           = [
+    yamlencode(merge(jsondecode(var.env0_values), jsondecode(var.customer_values)))
```
Comment (RLRabinowitz) [inline README.md:88]: This var doesn't exist in any file here. Either add some explanation here or add it to variables.tf or something

**Review by RLRabinowitz — COMMENTED** [review]
Looks good. Did you test it BTW?

Left some suggestions for making wording clearer in the README (which is basically the most important part of this IMO)

File: aws/csi-driver/main.tf [hunk 12]
```diff
@@ -0,0 +1,145 @@
+data "aws_eks_cluster" "cluster" {
```
Comment (RLRabinowitz) [inline aws/csi-driver/main.tf:1]: Consider adding README for each module, or at least some basic explanation on what's the purpose of each sub-module here

File: README.md (hunk 11 above)
Comment (HeverFarber) [reply README.md:88]: 342f65e

File: README.md (hunk 10 above)
Comment (HeverFarber) [reply README.md]: 342f65e

File: README.md (hunk 9 above)
Comment (HeverFarber) [reply README.md]: 342f65e

File: README.md (hunk 8 above)
Comment (HeverFarber) [reply README.md]: 342f65e

File: README.md (hunk 7 above)
Comment (HeverFarber) [reply README.md]: 342f65e

File: README.md (hunk 6 above)
Comment (HeverFarber) [reply README.md]: 342f65e

File: README.md (hunk 5 above)
Comment (HeverFarber) [reply README.md]: 342f65e

File: aws/csi-driver/main.tf (hunk 12 above)
Comment (HeverFarber) [reply aws/csi-driver/main.tf:1]: I think for now it redundant

**Review by RLRabinowitz — APPROVED** [review]
👏🏻

**Pull request merged**